File: trunk/src/java/org/codehaus/testdox/intellij/panel/TestElementCellRenderer.java
package org.codehaus.testdox.intellij.panel;

import java.awt.Component;

import javax.swing.JTable;
import javax.swing.table.DefaultTableCellRenderer;

import org.codehaus.testdox.intellij.TestElement;

public class TestElementCellRenderer extends DefaultTableCellRenderer {

    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus,
                                                   int row, int column) {
        TestElement element = (TestElement) value;
        setIcon(element.icon());
        return super.getTableCellRendererComponent(table, "<html>" + element.displayString() + "</html>", isSelected, false, row, column);
    }
}

File: trunk/src/main/scala/org/codehaus/testdox/intellij/ui/TestElementCellRenderer.scala
package org.codehaus.testdox.intellij.ui

import java.awt.Component

import javax.swing.JTable
import javax.swing.table.DefaultTableCellRenderer

import org.codehaus.testdox.intellij.TestElement

class TestElementCellRenderer extends DefaultTableCellRenderer {

  override def getTableCellRendererComponent(table: JTable, value: Any, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component = {
    val element = value.asInstanceOf[TestElement]
    setIcon(element.icon)
    super.getTableCellRendererComponent(table, "<html>" + element.displayString + "</html>", isSelected, false, row, column);
  }
}
